File: hoppity/c/hoppity.c
#include <stdio.h>
#include <stdlib.h>

#define BUFFERSIZE 50

int
main (int argc, char *argv[])
{
    long long int n;

    fscanf(fopen(argv[1],"r"), "%lld", &n);

    for (int i = 1; i <= n; i++) {
        if (i % 15 == 0) {
            printf("Hop\n");
        } else if (i % 5 == 0) {
            printf("Hophop\n");
        } else if (i % 3 == 0) {
            printf("Hoppity\n");
        }
    }

    return EXIT_SUCCESS;
}

File: hoppity/hoppity.py
#!/usr/bin/env python
import sys

filename = sys.argv[1]

inputfile = open(filename)

for i in range(1,1+int(inputfile.read())):
    if   i % 15 == 0: print 'Hop'
    elif i %  5 == 0: print 'Hophop'
    elif i %  3 == 0: print 'Hoppity'
